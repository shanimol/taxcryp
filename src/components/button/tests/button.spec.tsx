import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import Button from '../button';

configure({ adapter: new Adapter() });

const mockFn = jest.fn();

describe('Button Component', () => {
  it('Renders button', () => {
    const component = shallow(
      <Button
        onClick={mockFn}
        label="Save"
        className=''
      />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });

  it('Handles onClick', () => {
    const component = shallow(
      <Button
        onClick={mockFn}
        label="Save"
        className=''
      />
    );
    component.simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });
});