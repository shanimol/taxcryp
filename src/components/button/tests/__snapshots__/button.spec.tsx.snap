// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Renders button 1`] = `
<button
  className="btn  "
  onClick={[Function]}
  type="submit"
>
  Save
</button>
`;
